enhancements/hypershift/topology-aware-scheduling/topology-aware-scheduling.md

```diff
@@ -0,0 +1,203 @@
+it is suggested to install the operator and its managed components on the hosted-cluster.
+The following paragraphs detail the identified gaps and suggest resolution for each. 
+
+### Architectural Overview
+
+TAS solution comprises three main components:
+
+Resource-Topology-Exporter (RTE) - Responsible for collecting node information (mostly NUMA resources allocations) 
+and advertises it as a CR object named NRT (Node-Resource-Topology)
+
+Secondary-Scheduler - An add-on to the default scheduler which is responsible for
+scheduling the pod in the most suitable node according to the data exposed by the NRT object.
+
+NROP - The operator itself which is responsible for the deployment of RTE and the Scheduler.
+RTE and the scheduler are configurable via the NumaResourceOperator and NumaResourceScheduler CRs respectively.
+
+Further Reading: https://docs.google.com/presentation/d/14nVUJDDb5DJ9wJExGKJZe7evOuvkG3k14pI2K7wtLgM/edit#slide=id.g547716335e_0_260
+
+### Expose KubeletConfig to Hosted-Clusters
+
+NROP needs to learn the topology-manager (TM) scope and policy of the nodes on which TAS is running
+and feed RTE with that information.
+For that, NROP needs access to the KubeletConfig object which holds the information about the TM scope and policy.
+
+Since NROP is running on the hosted-cluster, it doesn’t have access to the KubeletConfig object.
+On HyperShift, the Hosted-Cluster-Config-Operator (HCCO) is responsible for rendering the KubeletConfig objects
+and applying their configuration on the node.
+It’s suggested that when HCCO renders KubeletConfig, it adds a ConfigMap that encapsulates the KubeletConfig data
+on the desired hosted clusters, so that NROP could access it and read the information it’s needed.
+The ConfigMap contains a NodePool label so NROP knows which group of nodes is associated with the given KubeletConfig.
+
+![Architecture Layout](architecture_layout.jpg)
+
+### Handle Node Grouping
+On HyperShift NROP maps the nodes of the hosted-cluster per NodePool.
+This is the equivalent of MachineConfigPool (MCP) on OCP.
+Each node has a label with a value of the NodePool it belongs to,
+so NROP iterates over the list of nodes and builds a map of [NodePoolName] []*corev1.Nodes.
+It is suggested to use NodeSelector instead of MachineConfigPoolSelector to have a unified API for both HCP and OCP.
+
+### Workflow Description
+
+The deployment and usage of NROP per-se, as a day2 operator,
+on HCP is similar to the deployment and usage on OCP. 
+No major changes are necessary on that front.
+
+### API Extensions
+
+Extend the [NUMAResourceOperator](https://github.com/openshift-kni/numaresources-operator/blob/0beeb62c7c0182ba5a2963ccd658bd0f6af44ebd/api/numaresourcesoperator/v1/numaresourcesoperator_types.go#L164) CR to contain NodeSelector filed.
+`spec.NodeGroup[].NodeSelector`
+User can choose the group of node by the NodeSelector and to have a unified API for OCP and HCP.
```

Comment:
I see no problem. We just need the extra validation logic to prevent incoherent configuration (user setting both fields)